### PR TITLE
Support custom boxes

### PR DESCRIPTION
**Pull request by yamgent**
**What is the purpose of this pull request? (put "X" next to an item, remove the rest)**

• [X] Enhancement to an existing feature

Fixes https://github.com/MarkBind/markbind/issues/69
Fixes https://github.com/MarkBind/markbind/issues/62

**What is the rationale for this request?**

User wants a way to customize tip-boxes with custom colors and icons. User also wants an empty-fill type tip boxes.

**What changes did you make? (Give an overview)**

Added new attributes `color`, `background-color`, `icon`, `border-color` for tip-box customization. Also added a new attribute `border-left-color` to support empty-fill type tip boxes.

**Provide some example code that this change will affect:**

```html
<tip-box type="info" icon=":rocket:">
    info but with rocket
</tip-box>
<tip-box type="info" color="white" background-color="red">
    info but with different colors
</tip-box>
<tip-box color="white" background-color="blue" border-color="darkblue">
    default with custom colors
</tip-box>            
<tip-box background-color="white" border-color="grey" border-left-color="blue">
    empty box with left border
</tip-box>
```

**Is there anything you'd like reviewers to focus on?**

Documentation wordings.

**Testing instructions:**

The documentation provides the code under the tip-box section, can use that web page to experiment with the new attributes shown in the documentation.

### CODE REVIEW

File: docs/example/tipBoxDocs.vue [hunk 1]
```diff
@@ -25,6 +26,18 @@
...
+            </tip-box>
+            <tip-box type="info" color="white" background-color="red">
+                info but with different colors
+            </tip-box>
+            <tip-box color="white" background-color="blue" border-color="darkblue">
+                default with custom colors
+            </tip-box>            
+            <box background-color="white" border-color="grey" border-left-color="blue">
+                empty box with left border
+            </box>
```
Comment (acjh) [inline docs/example/tipBoxDocs.vue]: - Let's move these into a separate `<div class="bs-example"></div><doc-code></doc-code>` group.
  At the moment, we need to scroll to see the doc-code.
- Two examples are sufficient: empty box (demo colours), info rocket (demo icon)
- Both can use `<box>`.
- Customise `type=default` before `type=info`.
- Be descriptive:
  - `empty box with left border` should be `default, styled as empty box with blue left border`,
  - `info but with rocket` should be `info, with rocket icon`,

File: docs/example/tipBoxDocs.vue [hunk 2]
```diff
@@ -51,7 +64,57 @@
...
+            <tip-box color="white" background-color="blue" border-color="darkblue">
+                default with custom colors
+            </tip-box>                       
+            <box background-color="white" border-color="grey" border-left-color="blue">
+                empty box with left border
+            </box>
         </doc-code>
+        <doc-table name="TipBox Options">
+          <div>
+            <p>backgroundColor</p>
```
Comment (acjh) [inline docs/example/tipBoxDocs.vue]: `background-color` (similar for others)

File: docs/example/tipBoxDocs.vue [hunk 3]
```diff
@@ -51,7 +64,57 @@
...
+          </div>
+          <div>
+            <p>icon</p>
+            <p><code>String</code></p>
+            <p><code>null</code></p>
+            <p>Icon to be used inside.</p>
+          </div>
+          <div>
+            <p>type</p>
+            <p><code>String</code>, one of <code>warning</code>, <code>info</code>, <code>definition</code>, <code>success</code>, <code>tip</code>, <code>important</code>, <code>wrong</code>, or empty for default.</p>
```
Comment (acjh) [inline docs/example/tipBoxDocs.vue]: - Use the same order as examples ("or empty for default." can remain at the end).
- Considering the length, let's move this to Description (as done in Panel Options).

File: docs/example/tipBoxDocs.vue [hunk 4]
```diff
@@ -51,7 +64,57 @@
...
+            <p>borderColor</p>
+            <p><code>String</code></p>
+            <p><code>null</code></p>
+            <p>Color of the entire border.</p>
+          </div>
+          <div>
+            <p>borderLeftColor</p>
+            <p><code>String</code></p>
+            <p><code>null</code></p>
+            <p>Color of the left border.</p>
```
Comment (acjh) [inline docs/example/tipBoxDocs.vue]: Leave description blank unless it adds value.
- Change this to "Overrides \`border-color\` for the left border."
- Also keep for `color`.

**Comment by yamgent** [comment]
@acjh thanks for the reviews!

**Comment by yamgent** [comment]
Changes made.

File: src/index.js [hunk 5]
```diff
@@ -30,6 +31,7 @@ const components = {
   accordion,
   affix,
   aside,
+  box,
```
Comment (acjh) [inline src/index.js]: Does `box: tipBox` à la `tabs: tabSet` work?

Comment (acjh) [reply src/index.js]: @yamgent In case you missed this.

**Pull request merged**